entity/comment.go

```diff
@@ -0,0 +1,12 @@
+package entity
+
+import "gorm.io/gorm"
+
+type Comment struct {
+	gorm.Model
+	Content string `gorm:"NOT NULL"`
+	UserID  uint
```

Comment:
Is this field are meant to be nullable? if yes, this could cause an type error. In Go in particular, null / empty values are different for each type, i.e, string with empty double quotes "", int with 0, struct with nil, etc. In this case, gorm will treat it as if the field are not nullable and will try to pass nil value when its null in the database. Therefore to fix this problem, you could set it to *uint so it can handle nill value 👍. You could read the explanation in the official gorm documentation here https://gorm.io/docs/models.html